pkg/imgpkg/imageset/unprocessed_image_refs.go

```diff
@@ -11,37 +11,50 @@ import (
 	regname "github.com/google/go-containerregistry/pkg/name"
 )
 
+const RootBundleLabelKey string = "dev.carvel.imgpkg.copy.root-bundle"
+
 type UnprocessedImageRef struct {
 	DigestRef string
 	Tag       string
+
+	Labels map[string]string
+}
+
+func (u UnprocessedImageRef) Key() string {
+	return u.DigestRef + u.Tag
 }
 
 type UnprocessedImageRefs struct {
-	imgRefs map[UnprocessedImageRef]struct{}
+	imgRefs map[string]UnprocessedImageRef
+
 	sync.Mutex
 }
 
 func NewUnprocessedImageRefs() *UnprocessedImageRefs {
-	return &UnprocessedImageRefs{imgRefs: map[UnprocessedImageRef]struct{}{}}
+	return &UnprocessedImageRefs{imgRefs: map[string]UnprocessedImageRef{}}
 }
 
 func (i *UnprocessedImageRefs) Add(imgRef UnprocessedImageRef) {
 	imgRef.Validate()
 
 	i.Mutex.Lock()
 	defer i.Mutex.Unlock()
-	i.imgRefs[imgRef] = struct{}{}
+	i.imgRefs[imgRef.Key()] = imgRef
 }
 
 func (i *UnprocessedImageRefs) Length() int {
 	return len(i.imgRefs)
 }
 
 func (i *UnprocessedImageRefs) All() []UnprocessedImageRef {
+	i.Mutex.Lock()
```

Comment:
while we are here, why do we need Mutex to be public here?

Reply:
for... transparency.... jk
made it private!